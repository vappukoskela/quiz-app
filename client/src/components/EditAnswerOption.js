import { Button, Checkbox, ListItem, ListItemIcon, TextField } from "@material-ui/core";
import DeleteIcon from '@material-ui/icons/Delete';
import { withStyles } from '@material-ui/core/styles';
import { green } from '@material-ui/core/colors';
import strings from '../localization/strings';

function EditAnswerOption(props) {
    const GreenCheckbox = withStyles({
        root: {
          color: green[400],
          '&$checked': {
            color: green[600],
          },
        },
        checked: {},
      })((props) => <Checkbox color="default" {...props} />);

    return (
        <ListItem key={"editansweropt" + props.value.id} role={undefined} dense >
            <ListItemIcon>
                <GreenCheckbox
                    onChange={(event) => props.updateAnsweroption(event, props.quiz, props.parentIndex, props.index, "CHECKBOX")}
                    checked={false}
                    edge="start"
                    tabIndex={-1}
                />
            </ListItemIcon>
            <ListItemIcon>
                <Checkbox
                    // TODO:
                    // onChange={(event) => console.log(event, "click")}
                     onChange={(event) => props.dispatch({ type: "SELECT_TOGGLE", data: { newText: event.target.value, quizIndex: props.quiz, questionIndex: props.parentIndex, answerIndex: props.index } })}
                    checked={props.value.selected}
                    edge="start"
                    tabIndex={-1}
                />
            </ListItemIcon>
                <div>
                    <TextField onChange={(event) => props.updateAnsweroption(event, props.quiz, props.parentIndex, props.index, "TEXT")} size="small" label={(strings.answer) + " " + (props.index + 1)} variant="outlined" value={props.value.answer} />
                    <Button className="deleteButton" onClick={(event) => props.deleteAnsweroption(event, props.quiz, props.parentIndex, props.index)}><DeleteIcon /></Button>
                </div>
        </ListItem>
    )
}

export default EditAnswerOption;