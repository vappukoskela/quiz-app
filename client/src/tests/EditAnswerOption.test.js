import React from 'react';
import renderer from 'react-test-renderer';
import EditAnswerOption from '../components/EditAnswerOption.js';

// Komponentti joka vaatii propsit renderöityäkseen oikein!

// FAIL: ei propseja. Checkboxin checked-value on undefined - ei renderöidy!
// it('renders correctly', () => {
//     const tree = renderer.create(<EditAnswerOptionu />).toJSON();
//     expect(tree).toMatchSnapshot();
// });

it('renders correctly with props', () => {
    const value = {id: 0, correct: false, selected: false}
    const tree = renderer.create(<EditAnswerOption value={value} />).toJSON();
    expect(tree).toMatchSnapshot();
});

it('renders correctly when checkbox correct', () => {
    const value = {id: 0, correct: true, selected: false}
    const tree = renderer.create(<EditAnswerOption value={value}  />).toJSON();
    expect(tree).toMatchSnapshot();
});

