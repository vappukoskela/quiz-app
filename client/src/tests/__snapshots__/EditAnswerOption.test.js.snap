// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly when checkbox correct 1`] = `
<li
  className="MuiListItem-root MuiListItem-dense MuiListItem-gutters"
  disabled={false}
>
  <div
    className="MuiListItemIcon-root"
  >
    <span
      aria-disabled={false}
      className="MuiButtonBase-root MuiIconButton-root PrivateSwitchBase-root-3 MuiCheckbox-root Component-root-11 MuiIconButton-edgeStart"
      onBlur={[Function]}
      onDragLeave={[Function]}
      onFocus={[Function]}
      onKeyDown={[Function]}
      onKeyUp={[Function]}
      onMouseDown={[Function]}
      onMouseLeave={[Function]}
      onMouseUp={[Function]}
      onTouchEnd={[Function]}
      onTouchMove={[Function]}
      onTouchStart={[Function]}
      tabIndex={null}
    >
      <span
        className="MuiIconButton-label"
      >
        <input
          checked={false}
          className="PrivateSwitchBase-input-6"
          data-indeterminate={false}
          onChange={[Function]}
          tabIndex={-1}
          type="checkbox"
        />
        <svg
          aria-hidden={true}
          className="MuiSvgIcon-root"
          focusable="false"
          viewBox="0 0 24 24"
        >
          <path
            d="M19 5v14H5V5h14m0-2H5c-1.1 0-2 .9-2 2v14c0 1.1.9 2 2 2h14c1.1 0 2-.9 2-2V5c0-1.1-.9-2-2-2z"
          />
        </svg>
      </span>
      <span
        className="MuiTouchRipple-root"
      />
    </span>
  </div>
  <div
    className="MuiListItemIcon-root"
  >
    <span
      aria-disabled={false}
      className="MuiButtonBase-root MuiIconButton-root PrivateSwitchBase-root-3 MuiCheckbox-root MuiCheckbox-colorSecondary MuiIconButton-colorSecondary MuiIconButton-edgeStart"
      onBlur={[Function]}
      onDragLeave={[Function]}
      onFocus={[Function]}
      onKeyDown={[Function]}
      onKeyUp={[Function]}
      onMouseDown={[Function]}
      onMouseLeave={[Function]}
      onMouseUp={[Function]}
      onTouchEnd={[Function]}
      onTouchMove={[Function]}
      onTouchStart={[Function]}
      tabIndex={null}
    >
      <span
        className="MuiIconButton-label"
      >
        <input
          checked={false}
          className="PrivateSwitchBase-input-6"
          data-indeterminate={false}
          onChange={[Function]}
          tabIndex={-1}
          type="checkbox"
        />
        <svg
          aria-hidden={true}
          className="MuiSvgIcon-root"
          focusable="false"
          viewBox="0 0 24 24"
        >
          <path
            d="M19 5v14H5V5h14m0-2H5c-1.1 0-2 .9-2 2v14c0 1.1.9 2 2 2h14c1.1 0 2-.9 2-2V5c0-1.1-.9-2-2-2z"
          />
        </svg>
      </span>
      <span
        className="MuiTouchRipple-root"
      />
    </span>
  </div>
  <div>
    <div
      className="MuiFormControl-root MuiTextField-root"
    >
      <label
        className="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-marginDense MuiInputLabel-outlined"
        data-shrink={false}
      >
        Answer NaN
      </label>
      <div
        className="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-formControl MuiInputBase-marginDense MuiOutlinedInput-marginDense"
        onClick={[Function]}
      >
        <input
          aria-invalid={false}
          autoFocus={false}
          className="MuiInputBase-input MuiOutlinedInput-input MuiInputBase-inputMarginDense MuiOutlinedInput-inputMarginDense"
          disabled={false}
          onAnimationStart={[Function]}
          onBlur={[Function]}
          onChange={[Function]}
          onFocus={[Function]}
          required={false}
          type="text"
        />
        <fieldset
          aria-hidden={true}
          className="PrivateNotchedOutline-root-7 MuiOutlinedInput-notchedOutline"
        >
          <legend
            className="PrivateNotchedOutline-legendLabelled-9"
          >
            <span>
              Answer NaN
            </span>
          </legend>
        </fieldset>
      </div>
    </div>
    <button
      className="MuiButtonBase-root MuiButton-root MuiButton-text deleteButton"
      disabled={false}
      onBlur={[Function]}
      onClick={[Function]}
      onDragLeave={[Function]}
      onFocus={[Function]}
      onKeyDown={[Function]}
      onKeyUp={[Function]}
      onMouseDown={[Function]}
      onMouseLeave={[Function]}
      onMouseUp={[Function]}
      onTouchEnd={[Function]}
      onTouchMove={[Function]}
      onTouchStart={[Function]}
      tabIndex={0}
      type="button"
    >
      <span
        className="MuiButton-label"
      >
        <svg
          aria-hidden={true}
          className="MuiSvgIcon-root"
          focusable="false"
          viewBox="0 0 24 24"
        >
          <path
            d="M6 19c0 1.1.9 2 2 2h8c1.1 0 2-.9 2-2V7H6v12zM19 4h-3.5l-1-1h-5l-1 1H5v2h14V4z"
          />
        </svg>
      </span>
      <span
        className="MuiTouchRipple-root"
      />
    </button>
  </div>
</li>
`;

exports[`renders correctly with props 1`] = `
<li
  className="MuiListItem-root MuiListItem-dense MuiListItem-gutters"
  disabled={false}
>
  <div
    className="MuiListItemIcon-root"
  >
    <span
      aria-disabled={false}
      className="MuiButtonBase-root MuiIconButton-root PrivateSwitchBase-root-3 MuiCheckbox-root Component-root-1 MuiIconButton-edgeStart"
      onBlur={[Function]}
      onDragLeave={[Function]}
      onFocus={[Function]}
      onKeyDown={[Function]}
      onKeyUp={[Function]}
      onMouseDown={[Function]}
      onMouseLeave={[Function]}
      onMouseUp={[Function]}
      onTouchEnd={[Function]}
      onTouchMove={[Function]}
      onTouchStart={[Function]}
      tabIndex={null}
    >
      <span
        className="MuiIconButton-label"
      >
        <input
          checked={false}
          className="PrivateSwitchBase-input-6"
          data-indeterminate={false}
          onChange={[Function]}
          tabIndex={-1}
          type="checkbox"
        />
        <svg
          aria-hidden={true}
          className="MuiSvgIcon-root"
          focusable="false"
          viewBox="0 0 24 24"
        >
          <path
            d="M19 5v14H5V5h14m0-2H5c-1.1 0-2 .9-2 2v14c0 1.1.9 2 2 2h14c1.1 0 2-.9 2-2V5c0-1.1-.9-2-2-2z"
          />
        </svg>
      </span>
    </span>
  </div>
  <div
    className="MuiListItemIcon-root"
  >
    <span
      aria-disabled={false}
      className="MuiButtonBase-root MuiIconButton-root PrivateSwitchBase-root-3 MuiCheckbox-root MuiCheckbox-colorSecondary MuiIconButton-colorSecondary MuiIconButton-edgeStart"
      onBlur={[Function]}
      onDragLeave={[Function]}
      onFocus={[Function]}
      onKeyDown={[Function]}
      onKeyUp={[Function]}
      onMouseDown={[Function]}
      onMouseLeave={[Function]}
      onMouseUp={[Function]}
      onTouchEnd={[Function]}
      onTouchMove={[Function]}
      onTouchStart={[Function]}
      tabIndex={null}
    >
      <span
        className="MuiIconButton-label"
      >
        <input
          checked={false}
          className="PrivateSwitchBase-input-6"
          data-indeterminate={false}
          onChange={[Function]}
          tabIndex={-1}
          type="checkbox"
        />
        <svg
          aria-hidden={true}
          className="MuiSvgIcon-root"
          focusable="false"
          viewBox="0 0 24 24"
        >
          <path
            d="M19 5v14H5V5h14m0-2H5c-1.1 0-2 .9-2 2v14c0 1.1.9 2 2 2h14c1.1 0 2-.9 2-2V5c0-1.1-.9-2-2-2z"
          />
        </svg>
      </span>
    </span>
  </div>
  <div>
    <div
      className="MuiFormControl-root MuiTextField-root"
    >
      <label
        className="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-marginDense MuiInputLabel-outlined"
        data-shrink={false}
      >
        Answer NaN
      </label>
      <div
        className="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-formControl MuiInputBase-marginDense MuiOutlinedInput-marginDense"
        onClick={[Function]}
      >
        <input
          aria-invalid={false}
          autoFocus={false}
          className="MuiInputBase-input MuiOutlinedInput-input MuiInputBase-inputMarginDense MuiOutlinedInput-inputMarginDense"
          disabled={false}
          onAnimationStart={[Function]}
          onBlur={[Function]}
          onChange={[Function]}
          onFocus={[Function]}
          required={false}
          type="text"
        />
        <fieldset
          aria-hidden={true}
          className="PrivateNotchedOutline-root-7 MuiOutlinedInput-notchedOutline"
        >
          <legend
            className="PrivateNotchedOutline-legendLabelled-9"
          >
            <span>
              Answer NaN
            </span>
          </legend>
        </fieldset>
      </div>
    </div>
    <button
      className="MuiButtonBase-root MuiButton-root MuiButton-text deleteButton"
      disabled={false}
      onBlur={[Function]}
      onClick={[Function]}
      onDragLeave={[Function]}
      onFocus={[Function]}
      onKeyDown={[Function]}
      onKeyUp={[Function]}
      onMouseDown={[Function]}
      onMouseLeave={[Function]}
      onMouseUp={[Function]}
      onTouchEnd={[Function]}
      onTouchMove={[Function]}
      onTouchStart={[Function]}
      tabIndex={0}
      type="button"
    >
      <span
        className="MuiButton-label"
      >
        <svg
          aria-hidden={true}
          className="MuiSvgIcon-root"
          focusable="false"
          viewBox="0 0 24 24"
        >
          <path
            d="M6 19c0 1.1.9 2 2 2h8c1.1 0 2-.9 2-2V7H6v12zM19 4h-3.5l-1-1h-5l-1 1H5v2h14V4z"
          />
        </svg>
      </span>
    </button>
  </div>
</li>
`;
